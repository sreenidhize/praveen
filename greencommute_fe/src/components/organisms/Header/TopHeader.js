import React from 'react';
import PropTypes from 'prop-types';
import Location  from '../../atom/Location/Location';
import { Avatar, makeStyles } from "@material-ui/core";
import theme from '../../../utils/theme/theme';
import Grid from '@material-ui/core/Grid';
import MessageOutlinedIcon from '@material-ui/icons/MessageOutlined';
import NotificationsNoneOutlinedIcon from '@material-ui/icons/NotificationsNoneOutlined';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles({
    root: {
        color: theme.palette.secondary.main,
        width: "20px",
        height: "80px",
    },
    label: {
        fontSize: theme.typography.navtabs.fontSize,
        paddingLeft: "10px",
        position: "relative",
    },
    avatarMargin: {
        marginLeft: "10px",
    }
});

const TopHeader = (props) => {
    const { locationLabel, profileName,icon,profile } = props;
    const classes = useStyles();

    return (

        <Grid container alignItems="center" justify="space-between" data-testid="topheader-placeholder">
            <Grid item xs={3}>
                <Location icon={icon} label={locationLabel} />
            </Grid>
            <Grid item container xs={3} alignItems="center" justify="space-evenly">
                <MessageOutlinedIcon className={classes.root} />
                <NotificationsNoneOutlinedIcon className={classes.root} />
                <Avatar className={classes.avatarMargin} src={profile}></Avatar>
                <Typography variant="body1" className={classes.label} >{profileName}</Typography>
                <ExpandMoreIcon className={classes.root} />
            </Grid>
        </Grid>
    );
};

TopHeader.propTypes = {
    locationLabel: PropTypes.string,
    profileName: PropTypes.string,
}
	
export default TopHeader;   