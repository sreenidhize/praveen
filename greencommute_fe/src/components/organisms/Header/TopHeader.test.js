import React from 'react';
import TopHeader from './TopHeader'
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";


describe("TopHeader placeholder test", () => {
    it("Matches the snapshot", () => {
      const { container } = render(
        <TopHeader locationLabel="Hyderabad" profileName="Bob" />
      );
      expect(container).toMatchSnapshot();
    });
  
    test("test TopHeader render", () => {
      const { getByTestId } = render(<TopHeader  locationLabel="hyderabad" profileName="Bob"/>);
      expect(getByTestId("topheader-placeholder")).toBeInTheDocument();
    });
  });