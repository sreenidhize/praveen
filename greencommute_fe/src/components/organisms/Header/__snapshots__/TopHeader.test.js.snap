// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopHeader placeholder test Matches the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
    data-testid="topheader-placeholder"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <div>
        <h4
          class="MuiTypography-root makeStyles-label-4 MuiTypography-h4"
        >
          Hyderabad
        </h4>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-evenly MuiGrid-grid-xs-3"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-root-1"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M4 4h16v12H5.17L4 17.17V4m0-2c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2H4zm2 10h12v2H6v-2zm0-3h12v2H6V9zm0-3h12v2H6V6z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-root-1"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.9 2 2 2zm6-6v-5c0-3.07-1.63-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.64 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2zm-2 1H8v-6c0-2.48 1.51-4.5 4-4.5s4 2.02 4 4.5v6z"
        />
      </svg>
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatarMargin-3 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <p
        class="MuiTypography-root makeStyles-label-2 MuiTypography-body1"
      >
        Bob
      </p>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-root-1"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </div>
  </div>
</div>
`;
